Source/WebCore/css/CSSComputedStyleDeclaration.cpp

```diff
@@ -51,10 +51,21 @@ CSSComputedStyleDeclaration::CSSComputedStyleDeclaration(Element& element, bool
     , m_allowVisitedStyle(allowVisitedStyle)
 {
     StringView name = pseudoElementName;
-    if (name.startsWith(':'))
-        name = name.substring(1);
-    if (name.startsWith(':'))
+    auto isLegacy = name == ":before"_s || name == ":after"_s || name == ":first-letter"_s || name == ":first-line"_s;
+
+    // handle "marker" and "foobar"
+    if (!name.startsWith(':'))
+        name = nullptr;
+    // handle ":marker" and ":foobar"    
+    else if (!isLegacy && !name.startsWith("::"_s))
+        name = nullptr;
+    // handle legacy pseudo
+    else if (isLegacy)
         name = name.substring(1);
+    // handle normal case "::marker" and "::foobar"
+    else
+        name = name.substring(2);
+
```

Comment:
@anttijk This fixes the parsing issue but I'm not sure yet how to solve the getComputedStyle returned value (also the comment are here to explain my choice, I can remove them later if necessary).
What do you think should be done? Or should it be a separate issue?

Reply:
Context for the returned values in 
https://github.com/web-platform-tests/wpt/pull/37452#discussion_r1046510669

Reply:
As mentioned on Slack, returning nullptr here: https://github.com/WebKit/WebKit/blob/8d03c66477e27515ed83a15d6e9f66ec484f2898/Source/WebCore/css/ComputedStyleExtractor.cpp#L2490-L2491 will probably yield the correct return value.

That might be done through nulling out `m_element` or adding a new PseudoId::Invalid and handling it in that method.

Reply:
@nt1m 
I tried to find a way to nullify the element but without success.
I managed a point to have the desired result by touching Element.cpp, but I clearly nullified the style for the parent element itself. 😮‍💨 

So I need a bit of help here. 

Does `m_element` can be modified? or is it constant?

And where could I do something like `m_element = nullptr;`


The parsing is all good. This is just the last stretch where I"m hitting the wall.